panic: interface conversion: ast.Node is *ast.BinaryOperator, not *ast.CompoundStmt

Hi,

Trying to convert 'micromax' to go:

panic: interface conversion: ast.Node is *ast.BinaryOperator, not *ast.CompoundStmt

goroutine 1 [running]:
panic(0x654960, 0xc42043cdc0)
        /usr/lib/go-1.7/src/runtime/panic.go:500 +0x1a1
github.com/elliotchance/c2go/transpiler.transpileDoStmt(0xc4202cefc0, 0xc42012a120, 0xc4208b7bd0, 0xc4208b7be8, 0x7b9380, 0xc4202cefc0, 0x1, 0xc42055e0c0, 0x0, 0x1, ...)
        /data/go/src/github.com/elliotchance/c2go/transpiler/branch.go:528 +0x55d
github.com/elliotchance/c2go/transpiler.transpileToStmt(0x7b9380, 0xc4202cefc0, 0xc42012a120, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /data/go/src/github.com/elliotchance/c2go/transpiler/transpiler.go:302 +0x104b
github.com/elliotchance/c2go/transpiler.transpileToStmts(0x7b9380, 0xc4202cefc0, 0xc42012a120, 0x0, 0x0, 0x0, 0x0, 0x0)
        /data/go/src/github.com/elliotchance/c2go/transpiler/transpiler.go:252 +0x14e

micromax is: http://home.hccnet.nl/h.g.muller/umax4_8.c

p.s. this is a for-fun conversion, I would not spend too much time on it!

--- a/tests/do.c
+++ b/tests/do.c
@@ -3,7 +3,7 @@
 
 int main()
 {
-    plan(6);
+    plan(7);
 	
 	int i = 0;
 
@@ -29,5 +29,10 @@ int main()
 	} while ((i /= 10) > 0);
 	is_eq( i , 8 );
 	
+	diag("do without CompoundStmt");
+	int s = 1;
+	do s = s == 0 ? s+=2 : s++;while(s > 10);
+	is_eq(s , 1);
+
 	done_testing();
 }

--- a/tests/do.c
+++ b/tests/do.c
@@ -31,8 +31,8 @@ int main()
 	
 	diag("do without CompoundStmt");
 	int s = 1;
-	do s = s == 0 ? s+=2 : s++;while(s > 10);
-	is_eq(s , 1);
+	do s++; while(s < 10);
+	is_eq(s , 10);
 
 	done_testing();
 }

--- a/transpiler/branch.go
+++ b/transpiler/branch.go
@@ -525,7 +525,14 @@ func transpileDoStmt(n *ast.DoStmt, p *program.Program) (
 	forOperator.AddChild(nil)
 	forOperator.AddChild(nil)
 	forOperator.AddChild(nil)
-	c := n.Children()[0].(*ast.CompoundStmt)
+	var c *ast.CompoundStmt
+	if _, ok := n.Children()[0].(*ast.CompoundStmt); ok {
+		c = n.Children()[0].(*ast.CompoundStmt)
+	} else {
+		var newComp ast.CompoundStmt
+		newComp.AddChild(n.Children()[0])
+		c = &newComp
+	}
 	ifBreak := createIfWithNotConditionAndBreak(n.Children()[1])
 	c.AddChild(&ifBreak)
 	forOperator.AddChild(c)

